src/ray/raylet/scheduling_queue.h

```diff
@@ -116,21 +116,35 @@ class SchedulingQueue {
   void MoveTasks(std::unordered_set<TaskID> tasks, TaskState src_state,
                  TaskState dst_state);
 
+  class TaskQueue {
+  public:
+    TaskQueue() {}
+    bool Append(const TaskID &task_id, const Task &task);
+    bool Remove(const TaskID &task_id);
+    bool Remove(const TaskID &task_id, std::vector<Task> &removed_tasks);
+
+    const std::list<Task> &GetTasks() const;
+  private:
+    std::list<Task> list_;
```

Comment:
let's use more descriptive names, perhaps a `task_list_` and `task_map_`?

Reply:
makes sense.